src/main/java/org/torusresearch/torusutils/helpers/NodeUtils.java

```diff
@@ -0,0 +1,573 @@
+import org.torusresearch.torusutils.apis.responses.VerifierLookupResponse.VerifierLookupResponse;
+import org.torusresearch.torusutils.types.FinalKeyData;
+import org.torusresearch.torusutils.types.Metadata;
+import org.torusresearch.torusutils.types.NodesData;
+import org.torusresearch.torusutils.types.OAuthKeyData;
+import org.torusresearch.torusutils.types.SessionData;
+import org.torusresearch.torusutils.types.TorusUtilsExtraParams;
+import org.torusresearch.torusutils.types.VerifierParams;
+import org.torusresearch.torusutils.types.common.ImportedShare;
+import org.torusresearch.torusutils.types.common.KeyLookup.KeyLookupResult;
+import org.torusresearch.torusutils.types.common.KeyLookup.KeyResult;
+import org.torusresearch.torusutils.types.common.PubNonce;
+import org.torusresearch.torusutils.types.common.SessionToken;
+import org.torusresearch.torusutils.types.common.TorusKey;
+import org.torusresearch.torusutils.types.common.TypeOfUser;
+import org.web3j.crypto.Hash;
+
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import okhttp3.internal.http2.Header;
+
+public class NodeUtils {
+    private NodeUtils() {
+    }
+
+    public static KeyLookupResult getPubKeyOrKeyAssign(@NotNull String[] endpoints, @NotNull Web3AuthNetwork network, @NotNull String verifier, @NotNull String verifierId, @NotNull String legacyMetdadataHost, @Nullable Integer serverTimeOffset, @Nullable String extendedVerifierId) throws Exception {
+        int threshold = (endpoints.length / 2) + 1;
+
+        BigInteger timeOffset = BigInteger.ZERO;
+        if (serverTimeOffset != null) {
+            timeOffset = BigInteger.valueOf(serverTimeOffset);
+        }
+        timeOffset = timeOffset.add( new BigInteger(String.valueOf(System.currentTimeMillis() / 1000)));
+
+        GetOrSetKeyParams params = new GetOrSetKeyParams(true, verifier, verifierId, extendedVerifierId, true, true, timeOffset.toString());
+        List<CompletableFuture<String>> lookupPromises = new ArrayList<>();
+        for (int i = 0; i < endpoints.length; i++) {
+            lookupPromises.add(i, APIUtils.post(endpoints[i], APIUtils.generateJsonRPCObject("GetPubKeyOrKeyAssign",
+                    params), true));
+        }
+
+        ArrayList<JsonRPCResponse<VerifierLookupResponse>> collected = new ArrayList<>();
+
+        JsonRPCErrorInfo errResult = null;
+        KeyResult key = null;
+        List<JsonRPCResponse<VerifierLookupResponse>> lookupPubKeys = null;
+        GetOrSetNonceResult nonce = null;
+
+        Gson json = new Gson();
+        for (CompletableFuture<String> lookup: lookupPromises) {
+            try {
+                String result = lookup.get();
```

Comment:
this is your replacement for some.. is it?

Reply:
Yes. Ideally should be Future instead of CompletableFuture, since the former is cancellable